### DISCUSSION TITLE
SwiftUI support in MAUI

### DISCUSSION THREAD

**Discussion by MihirJayavant** [dotnet/maui#254] [General]
SwiftUI is now well supported by Apple and is the future as many new ios and watchos functionality are exclusive to SwiftUI. I also see growing momentum towards swiftUI from ios developers which means many 3rd party packages will use swiftUI.

This also means swift language binding should be better than what is today in xamarin.macios project.

Currently, I don't see any reference or discussion regarding SwiftUI integration, Is this support planned for MAUI projects?

**Comment by motoko89** [comment]
This is my concern as well. More and more UI features will be SwiftUI only, especially for iOS-only features such as WatchApp or Widget extension and extensions in general. If Microsoft doesn't support this themselves, at least I hope they will provide extensibility support and guidance to 3rd-party devs who are eager to add functionalities for this

**Comment by GalaxiaGuy** [comment]
Not a complete answer:

SwiftUI support in Maui would need SwiftUI support in Xamarin.iOS/.NET iOS: https://github.com/xamarin/xamarin-macios/issues/6235

To make that work nicely there will need to be a good way to handle Swift associated types.

I have no idea what the status is, but there is: https://github.com/chkn/Xamarin.SwiftUI

**Reply by motoko89** [reply]
I keep seeing people pointing to chkn project but if you look at it, the last commit is 9 months ago! And even the author says it's not supposed to be used in production: `Until the project is ready for use in real apps, there are no official packages`

**Comment by davidortinau** [comment]
Can you provide some specific examples of what you want to use SwiftUI for via .NET MAUI? What are the "many 3rd party libraries" that you need for your application(s)?

**Reply by motoko89** [reply]
Hi David, for me I would like to:
1. Be able to create Widget extension through .NET MAUI, and be able to debug it through VS. Yes, there is the work around of writing the extension in XCode but it's so fragile for me and I can't attach debugger to it.
2. access new SwiftUI view in WatchOS

**Reply by scottkuhl** [reply]
The Journaling suggestion API is a good example that requires SwiftUI.  As far as I can tell, there is no way to support this in MAUI.

https://developer.apple.com/documentation/journalingsuggestions

**Comment by juepiezhongren** [comment]
swiftui is not mature enough for lob

**Comment by nbaulesglobalsolutions** [comment]
should i remove all iOS devices and accounts from the system?

Sent from my iPhone

On Jan 14, 2024, at 11:25 AM, Scott Kuhl ***@***.***> wrote:

﻿

The Journaling suggestion API is a good example that requires SwiftUI. As far as I can tell, there is no way to support this in MAUI.

https://developer.apple.com/documentation/journalingsuggestions
